bundles/org.openhab.binding.enocean/src/main/java/org/openhab/binding/enocean/internal/transceiver/EnOceanTransceiver.java

```diff
@@ -313,13 +313,19 @@ protected int read(byte[] buffer, int length) {
         InputStream localInputStream = inputStream;
         if (localInputStream != null) {
             try {
-                localInputStream.read(buffer, 0, length);
+                return localInputStream.read(buffer, 0, length);
             } catch (IOException e) {
                 logger.debug("IOException occured while reading the input stream", e);
                 return 0;
             }
+        } else {
+            logger.error("Cannot read from null stream");
```

Comment:
Can we reduce to **warn**? What it the user impact, will the binding stop working at this point?

Reply:
In fact yes, the binding should stop working at this point. This was not a recoverable error.
But, I said "was", because by reading the code I found that calling errorOccured allows the binding to know that it crashed and that it can try to restart.
So I made the modification, and warn is good.